ledger/acctupdates.go

```diff
@@ -1487,6 +1493,44 @@ func (au *accountUpdates) upgradeDatabaseSchema3(ctx context.Context, tx *sql.Tx
 	return 4, nil
 }
 
+// upgradeDatabaseSchema4 does not change the schema but migrates data:
+// remove empty AccountData entries from accountbase table
+func (au *accountUpdates) upgradeDatabaseSchema4(ctx context.Context, tx *sql.Tx) (updatedDBVersion int32, err error) {
+
+	queryAddresses := au.balancesTrie != nil
+	numDeleted, addresses, err := removeEmptyAccountData(tx, queryAddresses)
+	if err != nil {
+		return 0, err
+	}
+
+	if queryAddresses {
+		var totalHashesDeleted int
+		for _, addr := range addresses {
+			hash := accountHashBuilder(addr, basics.AccountData{}, []byte{0x80})
+			deleted, err := au.balancesTrie.Delete(hash)
```

Comment:
Please do double-check me, but I believe that the `au.balancesTrie` here would always be nil. in order to address this, do the following:
```go
if queryAddresses {
    mc, err0 := MakeMerkleCommitter(tx, false)
    trie, err := merkletrie.MakeTrie(mc, TrieMemoryConfig)
    for _, addr := range addresses {
       ... 
       trie.Delete(hash)
    }
    au.Commit()
}
```

Reply:
Thank you, you are right, it is created right after migrations...